scripts/install.ts

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable no-console */
+import {resolve} from "path";
+import {Readable} from "stream";
+import {finished} from "stream/promises";
+import {ReadableStream} from "stream/web";
+import {execSync} from "child_process";
+import {copyFileSync, createWriteStream, existsSync, mkdirSync} from "fs";
+import {exec, getBinaryName, getBindingsPath} from "../utils";
+
+const ROOT_DIR = resolve(__dirname, "..");
+const PREBUILD_DIR = resolve(ROOT_DIR, "prebuild");
+
+// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
+const VERSION: string = require(resolve(ROOT_DIR, "package.json")).version;
+
+/**
+ * Loading prebuilt bindings may fail in any number of unhappy ways, including a
+ * segfault. We use child processes to catch these unrecoverable process-level
+ * errors and continue the installation process
+ */
+async function testBindings(binaryPath: string): Promise<void> {
+  execSync(`node -e 'require("${binaryPath}")'`);
+}
+
+function getPrebuiltBinaryPath(binaryName: string): string {
+  return resolve(PREBUILD_DIR, binaryName);
+}
+
+function getReleaseUrl(binaryName: string): string {
+  return `https://github.com/ChainSafe/blst-ts/releases/download/v${VERSION}/${binaryName}`;
+}
+
+/**
+ * Download bindings from GitHub release
+ */
+async function downloadBindings(binaryName: string): Promise<string> {
+  const {body, status} = await fetch(getReleaseUrl(binaryName));
+
+  if (!body || status >= 400) {
+    throw new Error("Failed to download bindings");
+  }
+
+  if (!existsSync(PREBUILD_DIR)) {
+    mkdirSync(PREBUILD_DIR, {recursive: true});
+  }
+
+  const outputPath = getPrebuiltBinaryPath(binaryName);
+
+  await finished(Readable.fromWeb(body as ReadableStream<Uint8Array>).pipe(createWriteStream(outputPath)));
+
+  return outputPath;
+}
+
+async function buildBindings(binaryName: string): Promise<string> {
+  await exec("npm run clean:gyp", true, {cwd: ROOT_DIR});
+  await exec("npm run build:gyp", true, {cwd: ROOT_DIR});
+  const bindingPath = getBindingsPath(ROOT_DIR);
```

Comment:
after we use `node-gyp` do we know the exact location? looks like this `getBindingsPath` function is more suitable for other places not this one because it involves a search

Reply:
Most of the times yes but not always.

Because its possible to `npm install --debug` which will create a debug build in `build/Debug` instead of in `build/Release`.  So its necessary to check several places...